sbnci/Modules/TrackValidation/TrackValidation_module.cc

```
@@ -0,0 +1,3127 @@
+        //Getting track pitch collection
+        trkPitchYMap[caloPlaneId] = trkcalo->TrkPitchC();
+        //Getting the track XYZ 
+        //trkXYZYMap[caloPlaneId] = trkcalo->XYZ<std::vector<TVector3>>();
+        ++Nocaloloop;
+        //std::cout<<" No of calorimetry loop: "<<Nocaloloop<<std::endl;
+        //
+      }//close the loop over calorimetry
+      //std::cout<<" "<<trkdEdxMap.size()<<" "<<trkdQdxMap.size()<<" "<<trkKEMap.size()<<" "<<trkRangeMap.size()<<std::endl;
+
+      //Check if user wants neutrino tracks.
+      if(fmtrkpf.isValid() && fUseNeutrinoTracks){
+        if(fmtrkpf.at(trk.key()).size() > 0){
+          if(pfpsMap.find(fmtrkpf.at(trk.key()).at(0)->Parent()) != pfpsMap.end()){
+            if(pfpsMap[fmtrkpf.at(trk.key()).at(0)->Parent()]->PdgCode() != 12 && 
+                pfpsMap[fmtrkpf.at(trk.key()).at(0)->Parent()]->PdgCode() != 14){
+              if(fVerbose > 0){
+                std::cout << "Removing the track as it is not daughter of the neutrino" << std::endl;
+              }
+              continue;
+            }
+          }
+        }
+      }//Close the check if user wants neutrino tracks 
+
+      //========================
+      //=== Track Validation ===
+      //========================
+
+      //Get the hits vector from the track
+      std::vector<art::Ptr<recob::Hit>> trkhits = fmtrkh.at(trk.key());
+      int NumberofHitsinRecoTrk = trkhits.size();
+      if(fVerbose > 1){std::cout << "track hits size: " <<NumberofHitsinRecoTrk<<std::endl;}
+      if(NumberofHitsinRecoTrk < fMinHitSize){
+        if(fVerbose > 0){
+          std::cout << "Removing track as it doesn't have enough hits" << std::endl;
+        }
+        continue;
+      }
+
+      //If fUseBestPlane is 2 Y plane, 3 best-plane from number of hits or 4 the best plane from energy deposited.
+      int BestPlaneID = FindingTrackBeastPlane(clockData, trkhits);
+      if(BestPlaneID > 2){
```

This should never be the case, it should throw an exception if this happens not ignore it

---

resolved